src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -148,14 +213,17 @@ const useFetchChannelHook = () => {
     if (!localDb) return;
     let anonymousChannel: ChannelData[] = [];
     try {
-      anonymousChannel = await AnonymousMessageRepo.getAllAnonymousChannels();
+      const timeStamp = StorageUtils.channelAnonTimeStamps.get();
+      anonymousChannel = await AnonymousMessageRepo.getAllAnonymousChannels(timeStamp as string);
     } catch (e) {
       console.log('error on getting anonymousChannel:', e);
     }
 
     try {
       await saveAllChannelData(anonymousChannel, 'ANONYMOUS');
       refresh('channelList');
+      const timestamp = new Date().toISOString();
+      StorageUtils.channelAnonTimeStamps.set(timestamp);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, the changes here introduce the use of a timestamp for fetching anonymous channels. This can potentially improve performance by reducing the amount of data fetched from the server. However, it's important to ensure that this change doesn't introduce issues with data consistency. Consider adding tests to verify that all expected channels are correctly fetched and saved.


<!-- This is an auto-generated comment by OSS CodeRabbit -->